waterdata/location_utils.py

```diff
@@ -0,0 +1,148 @@
+    """
+    Convert values for keys that contains codes to human readable names using the lookups
+    :param dict location:
+    :param dict code_lookups:
+    :param dict country_state_county_lookups:
+    :rtype: dict
+    """
+
+    def get_state_name (country_code, state_code):
+        """
+        Return the name of the state with country_code and state_code
+        :param str country_code:
+        :param str state_code:
+        :rtype: str
+        """
+        country_lookup = country_state_county_lookups.get(country_code, {})
+        state_lookup = country_lookup.get('state_cd', {})
+
+        return state_lookup.get(state_code, {}).get('name', state_code)
+
+    transformed_location = {}
+
+    country_code = location.get('country_cd')
+    state_code = location.get('state_cd')
+    county_code = location.get('county_cd')
+    district_code = location.get('district_cd')
+
+    for (key, value) in location.items():
+        if key == 'state_cd' and country_code and state_code:
+            transformed_value = get_state_name(country_code, state_code)
+
+        elif key == 'district_cd' and country_code and district_code:
+            transformed_value = get_state_name(country_code, district_code)
+
+        elif key == 'county_cd' and country_code and state_code and country_code:
+            country_lookup = country_state_county_lookups.get(country_code, {})
+            state_lookup = country_lookup.get('state_cd', {}).get(state_code, {})
+            county_lookup = state_lookup.get('county_cd', {})
+
+            transformed_value = county_lookup.get(county_code, {}).get('name', county_code)
+
+        elif key in code_lookups:
+            transformed_value = code_lookups.get(key).get(value, {}).get('name', value)
+
+        else:
+            transformed_value = value
+        transformed_location[key] = transformed_value
+
+    return transformed_location
+
+
+def get_capabilities(location_parameter_records):
+    """
+    Determine the parameters measured at the site.
+
+    :param iterable location_parameter_records: an iterable containing of location parameters
+    :return: USGS parameter codes measured at a site
+    :rtype: set
+
+    """
+    supported_params = set([parameter_record['parm_cd'] for parameter_record in location_parameter_records])
+    return supported_params
+
+
+def get_site_parameter(location_parameter_records, parameter_cd):
+    """
+    Determine the period of record and number of records for a
+    parameter being measured at this site. If the parameter code
+    value specified is not available at this site, None is returned.
+
+    :param iterable location_parameter_records: an iterable containing of location parameter
+    :param str parameter_cd: the USGS parameter code of interest
+    :return: a parameter's "start_date", "end_date", and "record_count" if available; dates are Python date objects
+    :rtype: collection.namedtuple or None
```

Comment:
I wonder if there is a standard way to say we are returning a Parameter namedtuple?

Reply:
I haven't been able to find a standard or best practice on it's own. I've been wondering if it would make more sense to document it like a class in it's own right (e.g. :rtype: waterdata.location_utils.Parameter or None). In many ways, namedtuples can be thought as something one would use in-lieu of creating a normal Python class.

Reply:
I like that!